testsuite/integration/elytron/src/test/java/org/wildfly/test/integration/elytron/ssl/UndertowTwoWaySslNeedClientAuthTestCase.java

```diff
@@ -130,6 +146,153 @@ public void testSendingNoClientCertificateFails() {
+        });
+    }
+
+    /**
+     * RESTEasy client loads SSL Context from Elytron client config.
+     * This SSL Context does not have truststore configured, so exception is expected.
+     */
+    @Test(expected = ProcessingException.class)
+    public void testResteasyElytronClientMissingTruststore() {
+        AuthenticationContext context = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-correct-truststore-missing.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        context.run(() -> {
+            ResteasyClientBuilder resteasyClientBuilder = (ResteasyClientBuilder) ClientBuilder.newBuilder();
+            ResteasyClient client = resteasyClientBuilder.build();
+            Response response = client.target(String.valueOf(securedRootUrl)).request().get();
+            Assert.assertEquals("Hello World!", response.readEntity(String.class));
+            Assert.assertEquals(200, response.getStatus());
+        });
+    }
+
+    /**
+     * Elytron client has configured truststore that does not contain server's certificate.
+     * Test will pass because Elytron config is ignored since different ssl context is specified on RESTEasy client builder specifically.
+     */
+    @Test
+    public void testClientConfigProviderSSLContextIgnoredIfDifferentIsSet() throws URISyntaxException, GeneralSecurityException {
+        AuthenticationContextConfigurationClient AUTH_CONTEXT_CLIENT =
+                AccessController.doPrivileged((PrivilegedAction<AuthenticationContextConfigurationClient>) AuthenticationContextConfigurationClient::new);
+
+        AuthenticationContext context = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-correct-truststore-missing.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        AuthenticationContext contextWithTruststore = doPrivileged((PrivilegedAction<AuthenticationContext>) () -> {
+            try {
+                URL config = getClass().getResource("wildfly-config-correct-truststore.xml");
+                return ElytronXmlParser.parseAuthenticationClientConfiguration(config.toURI()).create();
+            } catch (Throwable t) {
+                throw new InvalidAuthenticationConfigurationException(t);
+            }
+        });
+        SSLContext sslContext = AUTH_CONTEXT_CLIENT.getSSLContext(new URI(securedRootUrl.getHost()), contextWithTruststore);
```

Comment:
The value returned from URL.getHost() is not valid input to new URI(String str).